tests/integration/kubernetes/gha-run.sh

```diff
@@ -241,6 +241,66 @@ install_kata_tools_placeholder() {
 		"after CI picks up the gha yaml changes required to test that installation."
 }
 
+function update_worker_label(){
+	local workers
+	local label="$1"
+	workers="$(kubectl get nodes --no-headers | grep '\<worker\>' | awk '{ print $1 }')"
+	for nodename in $workers; do
+		if ! kubectl get "node/${nodename}" -o jsonpath='{.metadata.labels}' | \
+			grep "$label" >/dev/null; then
+			kubectl label node "$nodename" --overwrite "${label}"
+		fi
+	done
+}
+
+function deploy_snapshotter() {
+	echo "::group::Deploying ${SNAPSHOTTER}"
+
+	case ${SNAPSHOTTER} in
+		nydus) deploy_nydus_snapshotter ;;
+		*) >&2 echo "${SNAPSHOTTER} flavour is not supported"; exit 2 ;;
+	esac
+
+	echo "::endgroup::"
+}
+
+function deploy_nydus_snapshotter() {
+    echo "::group::deploy_nydus_snapshotter"
+
+	local nydus_snapshotter_install_dir="/tmp/nydus-snapshotter"
+	if [ -d "${nydus_snapshotter_install_dir}" ]; then
+		rm -rf "${nydus_snapshotter_install_dir}"
+	fi
+	mkdir -p "${nydus_snapshotter_install_dir}"
+	nydus_snapshotter_version=$(get_from_kata_deps "externals.nydus-snapshotter.version")
+
+	pushd "$nydus_snapshotter_install_dir"
+	# Enable guest pull feature in nydus snapshotter
+	sed -i -e "s|FS_DRIVER: .*|FS_DRIVER: \"proxy\"|g" misc/snapshotter/nydus-snapshotter.yaml
+	# Disable to read snapshotter config from configmap
+	sed -i -e "s|ENABLE_CONFIG_FROM_VOLUME: .*|ENABLE_CONFIG_FROM_VOLUME: \"false\"|g" misc/snapshotter/nydus-snapshotter.yaml
+	# Enable to run snapshotter as a systemd service
+	sed -i -e "s|ENABLE_SYSTEMD_SERVICE: .*|ENABLE_SYSTEMD_SERVICE: \"true\"|g" misc/snapshotter/nydus-snapshotter.yaml
+	# Enable "runtime specific snapshotter" feature in containerd when configuring containerd for snapshotter
+	sed -i -e "s|ENABLE_RUNTIME_SPECIFIC_SNAPSHOTTER: .*|ENABLE_RUNTIME_SPECIFIC_SNAPSHOTTER: \"true\"|g" misc/snapshotter/nydus-snapshotter.yaml
+
+	# Deploy nydus snapshotter as a daemonset
+	kubectl create -f "nydus-snapshotter-rbac.yaml"
+	kubectl create -f "nydus-snapshotter.yaml"
+	popd
+
+	update_worker_label "dragonfly/nydus-snapshotter=true"
+	
+	echo "::endgroup::"
+	echo "::group::nydus snapshotter logs"
+	kubectl logs nydus-snapshotter -n nydus-system
+	kubectl describe pod nydus-snapshotter -n nydus-system
+	echo "::endgroup::"
+	echo "::group::containerd config"
+	cat /etc/containerd/config.toml
```

Comment:
Notice that this will print the local containerd configuration.  If you have nydus installed on AKS, for example, you want to print the file content on the remote nodes.

You can use `get_one_kata_node()`(https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/tests_common.sh#L59) to pick one node then `exec_host()`(https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/tests_common.sh#L71) to `cat /etc/containerd/config.toml` on that node.  One example where those functions are used: https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/k8s-volume.bats#L18

Reply:
Thanks @wainersm , but I'm not an expert on it. Could you please help me make it happen in another PR? I would remove it from this PR. The reason why we need this is that before deploying kata for CoCo, we need to set the label that indicates whether the snapshotter is installed or not. This way, the process of deploying kata can recognize the snapshotter status and configure containerd for snapshotter in each runtimeclass.

Reply:
@ChengyuZhu6 sure, we can work together on a follow up PR to do that.